### DISCUSSION TITLE
Interial Forces and Velocity Aux

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#22096] [Q&A Modules: Solid mechanics] [answered]
The below input file snippet is from `pull_and_shear.i`. I am working on a dynamics problem and I was looking at some tests. I see here that the NewmarkVelAux as well as the acceleration aux is only run on `timestep_end`. Is this just for this test? For an implicit problem, I need this turn run on Linear?

```
  [inertia_x] # M*accel + eta*M*vel
    type = InertialForce
    use_displaced_mesh = false
    variable = disp_x
    velocity = vel_x
    acceleration = accel_x
  []

  [vel_x] # Calculates and stores velocity at the end of the time step
    type = NewmarkVelAux
    variable = vel_x
    acceleration = accel_x
    execute_on = timestep_end
  []
```

**Comment by maxnezdyur** [comment, marked as answer]
Never mind, I see that IntertialForce computes the current velocity and acceleration within its own kernel. I mistakenly thought that it used the current velocity and acceleration from the AuxKernel.

**Reply by recuero** [reply]
That's correct.
